agimus_controller/tests/test_ocp_croco_joint_state.py

```diff
@@ -0,0 +1,166 @@
+        trajectory_points = []
+
+        for t in range(1, self._horizon_size):
+            fraction = t / self._horizon_size
+            oscillation = amplitude * np.sin(2 * np.pi * frequency * fraction)
+            q_t = q0.copy()
+            q_t[3] += oscillation
+            self._state_warmstart.append(
+                np.concatenate((q_t, np.zeros(self.robot_model.nv)))
+            )
+            qdot_t = (
+                np.zeros(self.robot_model.nv)
+                if t == 1
+                else self._state_warmstart[-1][: self.robot_model.nq]
+                - self._state_warmstart[t - 2][: self.robot_model.nq]
+            )
+            control_warmstart.append(
+                pin.rnea(
+                    self.robot_model,
+                    self.robot_data,
+                    q_t,
+                    qdot_t,
+                    np.zeros(self.robot_model.nv),
+                )
+            )
+
+            trajectory_points.append(
+                WeightedTrajectoryPoint(
+                    TrajectoryPoint(
+                        robot_configuration=self._state_warmstart[-1][
+                            : self.robot_model.nq
+                        ],
+                        robot_velocity=self._state_warmstart[-1][
+                            self.robot_model.nq : self.robot_model.nq
+                            + self.robot_model.nv
+                        ],
+                    ),
+                    TrajectoryPointWeights(
+                        w_robot_configuration=1e6,
+                    ),
+                )
+            )
+
+        # Solve OCP
+        self._state_reg = np.concatenate((q0, np.zeros(self.robot_model.nv)))
+        self._ocp = OCPCrocoJointState(self.robot_models, self._ocp_params)
+        self._ocp.set_reference_weighted_trajectory(trajectory_points)
+        self._ocp.solve(self._state_reg, self._state_warmstart, control_warmstart)
+
+        # Test solution consistency
+        for t in range(self._horizon_size - 1):
+            for joint_idx in range(self.robot_model.nq):
+                with self.subTest(t=t, joint=joint_idx):
+                    error = abs(
+                        self._state_warmstart[t][joint_idx]
+                        - self._ocp.ocp_results.states[t + 1][joint_idx]
+                    )
+                    self.assertLess(
+                        error,
+                        1e-2,
+                        f"Error at time {t}, joint {joint_idx}: {error:.5f}",
+                    )
+
+        # Test trajectories match the reference
+        reference_trajectory = [
+            self._state_warmstart[t][3] for t in range(self._horizon_size - 1)
+        ]
+        solution_trajectory = [
+            self._ocp.ocp_results.states[t + 1][3]
+            for t in range(self._horizon_size - 1)
+        ]
+        np.testing.assert_allclose(reference_trajectory, solution_trajectory, atol=1e-2)
+
+    def plot(self):
```

Comment:
I am wondering right now. Maybe there is an option to save plots as artefacts of unittests? Just so that if unittests filas we can simply download artifact from the runner as `.jpg` with a plot of results. What do you think @MaximilienNaveau  @jmirabel?

Reply:
I think we should not over-engineer it. Plots are useful for developers when writing test. I don't think they are useful for anything else.

If someone needs to see what is happening in CI because a test fails, then we can add what you suggest. Otherwise, this is not worth the effort.